src/main/java/uk/gov/companieshouse/insolvency/data/service/InsolvencyServiceImpl.java

```diff
@@ -40,68 +40,41 @@ public InsolvencyServiceImpl(Logger logger, InsolvencyRepository insolvencyRepos
                     insolvencyRepository.findById(companyNumber);
             OffsetDateTime dateFromBodyRequest = companyInsolvency.getInternalData().getDeltaAt();
 
-            if (insolvencyDocumentFromDbOptional.isPresent()) {
-                InsolvencyDocument insolvencyDocumentFromDb =
-                        insolvencyDocumentFromDbOptional.get();
-
-                OffsetDateTime deltaAtFromDbStr = insolvencyDocumentFromDb.getDeltaAt();
-
-                if (deltaAtFromDbStr == null || dateFromBodyRequest.isAfter(
-                        deltaAtFromDbStr)) {
-                    insolvencyDocument.setDeltaAt(dateFromBodyRequest);
-                    insolvencyDocument.setUpdatedAt(LocalDateTime.now());
-                    insolvencyDocument.getCompanyInsolvency().setStatus(null);
-
-                    insolvencyApiService.invokeChsKafkaApi(contextId, insolvencyDocument,
-                            EventType.CHANGED);
-
-                    logger.info(String.format("ChsKafka api CHANGED invoked "
-                                    + "updating successfully for context id %s "
-                                    + "and company number %s",
-                            contextId,
-                            companyNumber));
-
-                    insolvencyRepository.save(insolvencyDocument);
-                    logger.info(String.format(
-                            "Company insolvency is updated in MongoDB "
-                                    + "with context id %s and company number %s",
-                            contextId,
-                            companyNumber));
-                } else {
-                    logger.info("Insolvency not persisted as the record provided is older"
-                            + " than the one already stored.");
-                }
-            } else {
-                insolvencyDocument.setDeltaAt(dateFromBodyRequest);
-                insolvencyDocument.setUpdatedAt(LocalDateTime.now());
-                insolvencyDocument.getCompanyInsolvency().setStatus(null);
-
-                insolvencyApiService.invokeChsKafkaApi(contextId, insolvencyDocument,
-                        EventType.CHANGED);
-
-                logger.info(String.format("ChsKafka api CHANGED invoked "
-                                + "creating successfully for context id %s and company number %s",
-                        contextId,
-                        companyNumber));
+            insolvencyDocumentFromDbOptional
+                    .ifPresent(existingDocument -> {
+                                if (!dateFromBodyRequest.isAfter(existingDocument.getDeltaAt())) {
+                                    logger.info("Insolvency not persisted as the record provided is older"
+                                            + " than the one already stored.");
+                                    throw new IllegalArgumentException("Stale delta at");
```

Comment:
This changes the existing behaviour of a stale delta which is not within the scope of the ticket. Happy for you to leave it in though as long as it has been tested locally with the consumer and the change is documented on the ticket. It will also need regression tested in CIDEV.